### ISSUE TITLE
Make `sc.importbypath()` more robust

### ISSUE THREAD

**Issue by cliffckerr** [sciris/sciris#578]
Sometimes loads the wrong version, though hard to reproduce:
```py
import sciris as sc

ss2 = sc.importbypath('starsim2/starsim')
ss1 = sc.importbypath('starsim/starsim', 'ss1')

print(ss2.__version__)
print(ss1.__version__)
```

**Comment by cliffckerr** [comment]
This seems to work:
```py
import importlib.util
import sys

# Load mymodule1 from /path/to/mymodule1
mymodule1_path = '/home/cliffk/idm/starsim/starsim/__init__.py'
spec1 = importlib.util.spec_from_file_location("starsim1", mymodule1_path)
mymodule1 = importlib.util.module_from_spec(spec1)

# Temporarily insert starsim1 in sys.modules to resolve relative imports
sys.modules['starsim'] = mymodule1
sys.modules['starsim1'] = mymodule1
spec1.loader.exec_module(mymodule1)

# Load mymodule2 from /path/to/mymodule2
mymodule2_path = '/home/cliffk/idm/starsim2/starsim/__init__.py'
spec2 = importlib.util.spec_from_file_location("starsim2", mymodule2_path)
mymodule2 = importlib.util.module_from_spec(spec2)

# Temporarily insert starsim2 in sys.modules to resolve relative imports
sys.modules['starsim'] = mymodule2
sys.modules['starsim2'] = mymodule2
spec2.loader.exec_module(mymodule2)

# Clean up the sys.modules entries afterward if necessary
del sys.modules['starsim']
del sys.modules['starsim1']
del sys.modules['starsim2']

# Now you can use both versions of the module
s1 = mymodule1.Sim(diseases='sir', networks='random').run()
s2 = mymodule2.Sim(diseases='sir', networks='random').run()
```

**Comment by cliffckerr** [comment]
Closed by #613
